src/screens/Transfer/index.tsx

```diff
@@ -0,0 +1,183 @@
+import Button from "../../components/Button";
+import {RootState} from "../../redux/reducers";
+import {useRoute} from "@react-navigation/native";
+import {updateStockTransfer} from "../../redux/actions/transfers";
+import showPopup from "../../components/Popup";
+import {searchLocationByLocationNumber} from "../../redux/actions/locations";
+
+
+const Transfer = () => {
+    const route = useRoute();
+    const {item}: any = route.params
+    const dispatch = useDispatch();
+    const location = useSelector((rootState: RootState) => rootState.mainReducer.currentLocation)
+    const [state, setState] = useState<any>({
+        binToLocation: "",
+        binToLocationData: "",
+        quantity: "0",
+    })
+    const binLocationSearchQueryChange = (query: string) => {
+        setState({
+            ...state,
+            binLocationSearchQuery: query,
+        });
+        onBinLocationBarCodeSearchQuerySubmitted();
+    };
+
+    const onBinLocationBarCodeSearchQuerySubmitted = () => {
+        if (!state.binToLocation) {
+            showPopup({
+                message: 'Search query is empty',
+                positiveButton: {
+                    text: 'Ok',
+                },
+            });
+            return;
+        }
+
+        const actionCallback = (locationData: any) => {
+            if (!locationData || locationData.error) {
+                showPopup({
+                    message:
+                        'Bin Location not found with LocationNumber:' +
+                        state.binLocationName,
+                    positiveButton: {
+                        text: 'Ok',
+                    },
+                });
+                setState({
+                    ...state,
+                    binToLocation: '',
+                    binLocationSearchQuery: '',
+                });
+            } else if (locationData) {
+                setState({
+                    ...state,
+                    binToLocationData: locationData,
+                    binLocationSearchQuery: '',
+                });
+            }
+        };
+        dispatch(
+            searchLocationByLocationNumber(state.binToLocation, actionCallback),
+        );
+    };
+    const onChangeBin = (text: string) => {
+        setState({...state, binToLocation: text})
+    }
+    const onChangeQuantity = (text: string) => {
+        setState({...state, quantity: text})
+    }
+
+
+    const onTransfer = () => {
+        const request: any = {
+            "status": "COMPLETED",
+            "stockTransferNumber": "",
+            "description": "",
+            "origin.id": location.id,
+            "destination.id": location.id,
+            "stockTransferItems": [{
+                "product.id": item.inventoryItem.product.id,
+                "inventoryItem.id": item.inventoryItem.id,
+                "location.id": location.id,
+                "originBinLocation.id": item?.binLocation.id ?? "Never",
```

Comment:
This should not have a default value. I think it should be ok to leave it as `item?.binLocation?.id`